packages/notebook/src/browser/service/notebook-service.ts

```diff
@@ -138,14 +131,15 @@ export class NotebookService implements Disposable {
      * It takes a few seconds for the plugin host to start so that notebook data providers can be registered.
      * This methods waits until the notebook provider is registered.
      */
-    protected async waitForNotebookProvider(type: string): Promise<SimpleNotebookProviderInfo | undefined> {
+    protected async waitForNotebookProvider(type: string): Promise<NotebookProviderInfo | undefined> {
         if (this.notebookProviders.has(type)) {
             return this.notebookProviders.get(type);
         }
-        const deferred = new Deferred<SimpleNotebookProviderInfo | undefined>();
+        await this.willUseNotebookSerializerEmitter.sequence(async listener => listener(type));
```

Comment:
Suggestion: Since we don't interrupt the execution of the emitter, doesn't it make more sense to use `await Promise.all(emitter.fire(type))`?